lang/py/src/avro/schema.py

```diff
@@ -815,21 +892,38 @@ def make_avsc_object(json_data, names=None):
     logical_type = None
     if 'logicalType' in json_data:
       logical_type = json_data.get('logicalType')
-      if logical_type != 'decimal':
+      if logical_type not in constants.SUPPORTED_LOGICAL_TYPE:
        raise SchemaParseException("Currently does not support %s logical type" % logical_type)
     if type in PRIMITIVE_TYPES:
-      if type == 'bytes':
-        if logical_type == 'decimal':
-          precision = json_data.get('precision')
-          scale = 0 if json_data.get('scale') is None else json_data.get('scale')
-          return BytesDecimalSchema(precision, scale, other_props)
+      if type == 'int' and logical_type == constants.DATE:
+        return DateSchema(other_props)
+      elif type == 'int' and logical_type == constants.TIME_MILLIS:
+        return TimeMillisSchema(
+          other_props=other_props
```

Comment:
this is just my personal preference: for short parameters, I'd tend to keep them in one line if they're not getting too long) so that the code is a little bit condensed. not a big deal. :)